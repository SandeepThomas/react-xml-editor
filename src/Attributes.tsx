import * as React from 'react';

import Attribute from './Attribute';
import { Actions } from './types';
import { push } from './Util';

interface Props {
    actions: Actions;
    attributes: {[key: string]: string};
    id: string[];
    element: string;
}

export default class Attributes extends React.Component<Props> {
    public render(): React.ReactNode {
        const { actions, attributes, element, id } = this.props;
        return (
            <span className="attributes">
                { Object.keys(attributes).map((name: string) => (
                    <Attribute
                        actions={ actions }
                        element={ element }
                        id={ push(id, '$', name) }
                        key={ name }
                        name={ name }
                        value={ this.props.attributes[name] }
                    />
                )) }
            </span>
        );
    }
}
