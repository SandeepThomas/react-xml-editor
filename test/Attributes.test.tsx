import {describe, expect, jest, test} from '@jest/globals';

import * as React from 'react';
import renderer from 'react-test-renderer';

import Attributes from '../src/Attributes';

describe('Attributes component', () => {
    test('renders properly', () => {
        const actions = 'actions';
        const component = renderer.create(
            <Attributes 
                actions={actions as any}
                attributes={{a: 'b'}}
                id="id"
            />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});