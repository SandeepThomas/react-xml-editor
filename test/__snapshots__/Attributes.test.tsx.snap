// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attributes component renders properly 1`] = `
<div>
  <span
    class="attributes"
  >
    <span
      class="attribute"
    >
      <span
        class="punc"
      >
         
      </span>
      <span
        class="name"
      >
        a
      </span>
      <span
        class="punc"
      >
        =
      </span>
      <span
        class="valueContainer"
      >
        <span
          class="punc"
        >
          "
        </span>
        <span
          class="value"
        >
          b
        </span>
        <span
          class="punc"
        >
          "
        </span>
      </span>
    </span>
  </span>
</div>
`;
